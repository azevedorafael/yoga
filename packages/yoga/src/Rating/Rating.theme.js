const Rating = ({ spacing, colors }) => ({
  backgroundColor: colors.rating,
  icon: {
    size: 12,
  },
  gutter: spacing.xxsmall,
});

export default Rating;
