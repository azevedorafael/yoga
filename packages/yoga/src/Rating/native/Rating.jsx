import React from 'react';
import styled, { withTheme } from 'styled-components';
import { number, func } from 'prop-types';

import { Star } from '@gympass/yoga-icons';

const SVG_DEFAULT_SIZE = 12;

const RatingWrapper = styled.View`
  flex-direction: row;
  ${({
    width,
    theme: {
      yoga: {
        components: { rating },
      },
    },
  }) => `
    width: ${width}px;
    height: ${rating.icon.size}px;
  `}
`;

/** Use the Rating component to view other people's opinions and experiences. */
const Rating = ({
  value,
  icon: Icon,
  max,
  theme: {
    yoga: {
      colors: { gray },
      components: { rating },
    },
  },
  ...rest
}) => (
  <RatingWrapper
    width={rating.gutter * (max - 1) + rating.icon.size * max}
    {...rest}
  >
    {Array.from({ length: max }, (_, i) => {
      const diff = i + 1 - value;

      if (diff <= 0) {
        return (
          <Icon
            fill={rating.backgroundColor}
            key={`filled-${i}`}
            width={rating.icon.size}
            height={rating.icon.size}
            viewBox={`0 0 ${SVG_DEFAULT_SIZE} ${SVG_DEFAULT_SIZE}`}
            style={{
              marginLeft: i !== 0 ? rating.gutter : undefined,
            }}
          />
        );
      }

      if (diff > 0 && diff < 1) {
        const width = (1 - diff) * rating.icon.size;
        const dWidth = diff * rating.icon.size;
        const wViewBox = SVG_DEFAULT_SIZE * (1 - diff);
        const dViewBox = SVG_DEFAULT_SIZE * diff;

        return (
          <React.Fragment key={`half-${i}`}>
            <Icon
              fill={rating.backgroundColor}
              width={width}
              height={rating.icon.size}
              viewBox={`0 0 ${wViewBox} ${SVG_DEFAULT_SIZE}`}
              style={{
                marginLeft: i !== 0 ? rating.gutter : undefined,
              }}
            />
            <Icon
              fill={gray[5]}
              width={dWidth}
              height={rating.icon.size}
              viewBox={`${wViewBox} 0 ${dViewBox} ${SVG_DEFAULT_SIZE}`}
            />
          </React.Fragment>
        );
      }

      return (
        <Icon
          fill={gray[5]}
          key={`unfilled-${i}`}
          width={rating.icon.size}
          height={rating.icon.size}
          viewBox={`0 0 ${SVG_DEFAULT_SIZE} ${SVG_DEFAULT_SIZE}`}
          style={{
            marginLeft: i !== 0 ? rating.gutter : undefined,
          }}
        />
      );
    })}
  </RatingWrapper>
);

Rating.propTypes = {
  value: number,
  /** The icon to display */
  icon: func,
  /** Maximum rating */
  max: number,
};

Rating.defaultProps = {
  value: undefined,
  icon: Star,
  max: 5,
};

export default withTheme(Rating);
