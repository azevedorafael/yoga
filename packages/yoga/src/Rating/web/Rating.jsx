/* eslint-disable jsx-a11y/mouse-events-have-key-events */

import React, { useState, useRef } from 'react';
import styled, { withTheme } from 'styled-components';
import { number, func, shape, bool } from 'prop-types';
import { Star } from '@gympass/yoga-icons';

import { max as maxPropType } from '../../shared';

const RatingWrapper = styled.div`
  display: inline-flex;

  :hover {
    cursor: pointer;
  }

  svg {
    pointer-events: none;
  }

  ${({ width, height, readOnly }) => `
    width: ${width}px;
    height: ${height}px;

    ${readOnly ? 'pointer-events: none;' : ''}
  `}
`;

/** Use the Rating component to view other people's opinions and experiences. */
const Rating = ({
  value,
  max,
  readOnly,
  onRate,
  onMouseOver,
  onMouseMove,
  onMouseLeave,
  theme: {
    yoga: {
      colors: { gray },
      components: { rating },
    },
  },
  icon: { type: Icon, size: iconSize = rating.icon.size },
  ...rest
}) => {
  const wrapperRef = useRef(null);
  const [hover, setHover] = useState(false);
  const [currentRating, setCurrentRating] = useState(0);

  return (
    <RatingWrapper
      width={rating.gutter * (max - 1) + iconSize * max}
      height={iconSize}
      mouseOver={hover}
      readOnly={readOnly}
      {...rest}
      onMouseOver={e => {
        setHover(true);
        onMouseOver(e);
      }}
      onMouseMove={e => {
        setCurrentRating(
          Math.ceil(
            (e.clientX - e.target.offsetLeft) /
              (wrapperRef.current.offsetWidth / max),
          ) || 1,
        );

        onMouseMove(e);
      }}
      onMouseLeave={e => {
        setHover(false);
        setCurrentRating(0);
        onMouseLeave(e);
      }}
      onClick={() => onRate(currentRating)}
      ref={wrapperRef}
    >
      {Array.from({ length: max }, (_, i) => {
        const diff = i + 1 - value;
        let width;

        if (currentRating >= i + 1 || (!hover && diff <= 0)) {
          return (
            <Icon
              fill={rating.backgroundColor}
              key={`filled-${i}`}
              width={iconSize}
              height={iconSize}
              viewBox={`0 0 ${iconSize} ${iconSize}`}
              style={{
                marginLeft: i !== 0 ? rating.gutter : undefined,
              }}
            />
          );
        }

        if (!hover && diff > 0 && diff < 1) {
          width = (1 - diff) * iconSize;
          const dWidth = diff * iconSize;
          const wViewBox = iconSize * (1 - diff);
          const dViewBox = iconSize * diff;

          return (
            <React.Fragment key={`half-${i}`}>
              <Icon
                fill={rating.backgroundColor}
                width={width}
                height={iconSize}
                viewBox={`0 0 ${wViewBox} ${iconSize}`}
                style={{
                  marginLeft: i !== 0 ? rating.gutter : undefined,
                }}
              />
              <Icon
                fill={gray[5]}
                width={dWidth}
                height={iconSize}
                viewBox={`${wViewBox} 0 ${dViewBox} ${iconSize}`}
                style={{
                  marginLeft: 'unset',
                }}
              />
            </React.Fragment>
          );
        }

        return (
          <Icon
            fill={gray[5]}
            key={`unfilled-${i}`}
            width={iconSize}
            height={iconSize}
            viewBox={`0 0 ${iconSize} ${iconSize}`}
            style={{
              marginLeft: i !== 0 ? rating.gutter : undefined,
            }}
          />
        );
      })}
    </RatingWrapper>
  );
};

Rating.propTypes = {
  value: number,
  /** The icon to display */
  icon: shape({
    type: func,
    size: maxPropType(24),
  }),
  /** Maximum rating */
  max: number,
  /** false to make it interactable */
  readOnly: bool,
  /** Event to be fired on click */
  onRate: func,
  onMouseOver: func,
  onMouseMove: func,
  onMouseLeave: func,
};

Rating.defaultProps = {
  value: undefined,
  icon: {
    type: Star,
    size: 12,
  },
  max: 5,
  readOnly: true,
  onRate: rating => {}, // eslint-disable-line no-unused-vars
  onMouseOver: () => {},
  onMouseMove: () => {},
  onMouseLeave: () => {},
};

export default withTheme(Rating);
