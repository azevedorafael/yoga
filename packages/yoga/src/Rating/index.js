import Rating from './web';

export default Rating;
