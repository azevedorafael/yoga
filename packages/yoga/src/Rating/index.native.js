import Rating from './native';

export default Rating;
