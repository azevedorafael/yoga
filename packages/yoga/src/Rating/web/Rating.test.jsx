import React from 'react';
import { render } from '@testing-library/react';
import { ThemeProvider } from '../..';
import Rating from '.';

describe('<Rating />', () => {
  describe('Snapshots', () => {
    it('should match snapshot with Rating', () => {
      const { container } = render(
        <ThemeProvider>
          <Rating />
        </ThemeProvider>,
      );
      expect(container).toMatchSnapshot();
    });

    it('should match snapshot with different values', () => {
      const { container } = render(
        <ThemeProvider>
          <Rating value={0} />
          <Rating value={0.5} />

          <Rating value={1} />
          <Rating value={1.1} />
          <Rating value={1.2} />
          <Rating value={1.3} />
          <Rating value={1.4} />
          <Rating value={1.5} />
          <Rating value={1.6} />
          <Rating value={1.7} />
          <Rating value={1.8} />
          <Rating value={1.9} />

          <Rating value={2} />
          <Rating value={2.5} />
          <Rating value={3} />
          <Rating value={3.5} />
          <Rating value={4} />
          <Rating value={4.5} />
        </ThemeProvider>,
      );
      expect(container).toMatchSnapshot();
    });

    it('should match snapshot with different icon quantity', () => {
      const { container } = render(
        <ThemeProvider>
          <Rating iconQuantity={3} value={0} />
          <Rating iconQuantity={3} value={0.5} />
          <Rating iconQuantity={3} value={1} />
          <Rating iconQuantity={3} value={1.5} />
          <Rating iconQuantity={3} value={2} />
          <Rating iconQuantity={3} value={2.5} />
          <Rating iconQuantity={3} value={3} />
        </ThemeProvider>,
      );
      expect(container).toMatchSnapshot();
    });

    it('should match snapshot with different icon', () => {
      const Circle = props => (
        <svg {...props}>
          <circle cx="6" cy="6" r="6" />
        </svg>
      );

      const { container } = render(
        <ThemeProvider>
          <Rating icon={Circle} value={0} />
          <Rating icon={Circle} value={1} />
          <Rating icon={Circle} value={2} />
        </ThemeProvider>,
      );
      expect(container).toMatchSnapshot();
    });
  });
});
