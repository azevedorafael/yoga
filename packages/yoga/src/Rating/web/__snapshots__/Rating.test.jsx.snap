// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Rating /> Snapshots should match snapshot when mouse over 1`] = `
.c0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  width: 76px;
  height: 12px;
}

.c0:hover {
  cursor: pointer;
}

.c0 svg {
  pointer-events: none;
}

.c0 svg {
  margin-left: 2px;
  margin-right: 2px;
}

<div>
  <div
    class="c0"
    height="12"
    width="76"
  >
    <svg
      fill="#FFC24C"
      height="12"
      viewBox="0 0 12 12"
      width="12"
    />
    <svg
      fill="#FFC24C"
      height="12"
      viewBox="0 0 12 12"
      width="12"
    />
    <svg
      fill="#FFC24C"
      height="12"
      viewBox="0 0 12 12"
      width="12"
    />
    <svg
      fill="#C5C5D6"
      height="12"
      viewBox="0 0 12 12"
      width="12"
    />
    <svg
      fill="#C5C5D6"
      height="12"
      viewBox="0 0 12 12"
      width="12"
    />
  </div>
</div>
`;

exports[`<Rating /> Snapshots should match snapshot with Rating 1`] = `
.c0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  width: 76px;
  height: 12px;
  pointer-events: none;
}

.c0:hover {
  cursor: pointer;
}

.c0 svg {
  pointer-events: none;
}

.c0 svg {
  margin-left: 2px;
  margin-right: 2px;
}

<div>
  <div
    class="c0"
    height="12"
    readonly=""
    width="76"
  >
    <svg
      fill="#C5C5D6"
      height="12"
      viewBox="0 0 12 12"
      width="12"
    />
    <svg
      fill="#C5C5D6"
      height="12"
      viewBox="0 0 12 12"
      width="12"
    />
    <svg
      fill="#C5C5D6"
      height="12"
      viewBox="0 0 12 12"
      width="12"
    />
    <svg
      fill="#C5C5D6"
      height="12"
      viewBox="0 0 12 12"
      width="12"
    />
    <svg
      fill="#C5C5D6"
      height="12"
      viewBox="0 0 12 12"
      width="12"
    />
  </div>
</div>
`;

exports[`<Rating /> Snapshots should match snapshot with different icon 1`] = `
.c0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  width: 76px;
  height: 12px;
  pointer-events: none;
}

.c0:hover {
  cursor: pointer;
}

.c0 svg {
  pointer-events: none;
}

.c0 svg {
  margin-left: 2px;
  margin-right: 2px;
}

<div>
  <div
    class="c0"
    height="12"
    readonly=""
    width="76"
  >
    <svg
      fill="#C5C5D6"
      height="12"
      viewBox="0 0 12 12"
      width="12"
    >
      <circle
        cx="12"
        cy="12"
        r="12"
      />
    </svg>
    <svg
      fill="#C5C5D6"
      height="12"
      viewBox="0 0 12 12"
      width="12"
    >
      <circle
        cx="12"
        cy="12"
        r="12"
      />
    </svg>
    <svg
      fill="#C5C5D6"
      height="12"
      viewBox="0 0 12 12"
      width="12"
    >
      <circle
        cx="12"
        cy="12"
        r="12"
      />
    </svg>
    <svg
      fill="#C5C5D6"
      height="12"
      viewBox="0 0 12 12"
      width="12"
    >
      <circle
        cx="12"
        cy="12"
        r="12"
      />
    </svg>
    <svg
      fill="#C5C5D6"
      height="12"
      viewBox="0 0 12 12"
      width="12"
    >
      <circle
        cx="12"
        cy="12"
        r="12"
      />
    </svg>
  </div>
  <div
    class="c0"
    height="12"
    readonly=""
    width="76"
  >
    <svg
      fill="#FFC24C"
      height="12"
      viewBox="0 0 12 12"
      width="12"
    >
      <circle
        cx="12"
        cy="12"
        r="12"
      />
    </svg>
    <svg
      fill="#C5C5D6"
      height="12"
      viewBox="0 0 12 12"
      width="12"
    >
      <circle
        cx="12"
        cy="12"
        r="12"
      />
    </svg>
    <svg
      fill="#C5C5D6"
      height="12"
      viewBox="0 0 12 12"
      width="12"
    >
      <circle
        cx="12"
        cy="12"
        r="12"
      />
    </svg>
    <svg
      fill="#C5C5D6"
      height="12"
      viewBox="0 0 12 12"
      width="12"
    >
      <circle
        cx="12"
        cy="12"
        r="12"
      />
    </svg>
    <svg
      fill="#C5C5D6"
      height="12"
      viewBox="0 0 12 12"
      width="12"
    >
      <circle
        cx="12"
        cy="12"
        r="12"
      />
    </svg>
  </div>
  <div
    class="c0"
    height="12"
    readonly=""
    width="76"
  >
    <svg
      fill="#FFC24C"
      height="12"
      viewBox="0 0 12 12"
      width="12"
    >
      <circle
        cx="12"
        cy="12"
        r="12"
      />
    </svg>
    <svg
      fill="#FFC24C"
      height="12"
      viewBox="0 0 12 12"
      width="12"
    >
      <circle
        cx="12"
        cy="12"
        r="12"
      />
    </svg>
    <svg
      fill="#C5C5D6"
      height="12"
      viewBox="0 0 12 12"
      width="12"
    >
      <circle
        cx="12"
        cy="12"
        r="12"
      />
    </svg>
    <svg
      fill="#C5C5D6"
      height="12"
      viewBox="0 0 12 12"
      width="12"
    >
      <circle
        cx="12"
        cy="12"
        r="12"
      />
    </svg>
    <svg
      fill="#C5C5D6"
      height="12"
      viewBox="0 0 12 12"
      width="12"
    >
      <circle
        cx="12"
        cy="12"
        r="12"
      />
    </svg>
  </div>
</div>
`;

exports[`<Rating /> Snapshots should match snapshot with different icon quantity 1`] = `
.c0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  width: 44px;
  height: 12px;
  pointer-events: none;
}

.c0:hover {
  cursor: pointer;
}

.c0 svg {
  pointer-events: none;
}

.c0 svg {
  margin-left: 2px;
  margin-right: 2px;
}

<div>
  <div
    class="c0"
    height="12"
    readonly=""
    width="44"
  >
    <svg
      fill="#C5C5D6"
      height="12"
      viewBox="0 0 12 12"
      width="12"
    />
    <svg
      fill="#C5C5D6"
      height="12"
      viewBox="0 0 12 12"
      width="12"
    />
    <svg
      fill="#C5C5D6"
      height="12"
      viewBox="0 0 12 12"
      width="12"
    />
  </div>
  <div
    class="c0"
    height="12"
    readonly=""
    width="44"
  >
    <svg
      fill="#FFC24C"
      height="12"
      viewBox="0 0 6 12"
      width="6"
    />
    <svg
      fill="#C5C5D6"
      height="12"
      viewBox="6 0 6 12"
      width="6"
    />
    <svg
      fill="#C5C5D6"
      height="12"
      viewBox="0 0 12 12"
      width="12"
    />
    <svg
      fill="#C5C5D6"
      height="12"
      viewBox="0 0 12 12"
      width="12"
    />
  </div>
  <div
    class="c0"
    height="12"
    readonly=""
    width="44"
  >
    <svg
      fill="#FFC24C"
      height="12"
      viewBox="0 0 12 12"
      width="12"
    />
    <svg
      fill="#C5C5D6"
      height="12"
      viewBox="0 0 12 12"
      width="12"
    />
    <svg
      fill="#C5C5D6"
      height="12"
      viewBox="0 0 12 12"
      width="12"
    />
  </div>
  <div
    class="c0"
    height="12"
    readonly=""
    width="44"
  >
    <svg
      fill="#FFC24C"
      height="12"
      viewBox="0 0 12 12"
      width="12"
    />
    <svg
      fill="#FFC24C"
      height="12"
      viewBox="0 0 6 12"
      width="6"
    />
    <svg
      fill="#C5C5D6"
      height="12"
      viewBox="6 0 6 12"
      width="6"
    />
    <svg
      fill="#C5C5D6"
      height="12"
      viewBox="0 0 12 12"
      width="12"
    />
  </div>
  <div
    class="c0"
    height="12"
    readonly=""
    width="44"
  >
    <svg
      fill="#FFC24C"
      height="12"
      viewBox="0 0 12 12"
      width="12"
    />
    <svg
      fill="#FFC24C"
      height="12"
      viewBox="0 0 12 12"
      width="12"
    />
    <svg
      fill="#C5C5D6"
      height="12"
      viewBox="0 0 12 12"
      width="12"
    />
  </div>
  <div
    class="c0"
    height="12"
    readonly=""
    width="44"
  >
    <svg
      fill="#FFC24C"
      height="12"
      viewBox="0 0 12 12"
      width="12"
    />
    <svg
      fill="#FFC24C"
      height="12"
      viewBox="0 0 12 12"
      width="12"
    />
    <svg
      fill="#FFC24C"
      height="12"
      viewBox="0 0 6 12"
      width="6"
    />
    <svg
      fill="#C5C5D6"
      height="12"
      viewBox="6 0 6 12"
      width="6"
    />
  </div>
  <div
    class="c0"
    height="12"
    readonly=""
    width="44"
  >
    <svg
      fill="#FFC24C"
      height="12"
      viewBox="0 0 12 12"
      width="12"
    />
    <svg
      fill="#FFC24C"
      height="12"
      viewBox="0 0 12 12"
      width="12"
    />
    <svg
      fill="#FFC24C"
      height="12"
      viewBox="0 0 12 12"
      width="12"
    />
  </div>
</div>
`;

exports[`<Rating /> Snapshots should match snapshot with different values 1`] = `
.c0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  width: 76px;
  height: 12px;
  pointer-events: none;
}

.c0:hover {
  cursor: pointer;
}

.c0 svg {
  pointer-events: none;
}

.c0 svg {
  margin-left: 2px;
  margin-right: 2px;
}

<div>
  <div
    class="c0"
    height="12"
    readonly=""
    width="76"
  >
    <svg
      fill="#C5C5D6"
      height="12"
      viewBox="0 0 12 12"
      width="12"
    />
    <svg
      fill="#C5C5D6"
      height="12"
      viewBox="0 0 12 12"
      width="12"
    />
    <svg
      fill="#C5C5D6"
      height="12"
      viewBox="0 0 12 12"
      width="12"
    />
    <svg
      fill="#C5C5D6"
      height="12"
      viewBox="0 0 12 12"
      width="12"
    />
    <svg
      fill="#C5C5D6"
      height="12"
      viewBox="0 0 12 12"
      width="12"
    />
  </div>
  <div
    class="c0"
    height="12"
    readonly=""
    width="76"
  >
    <svg
      fill="#FFC24C"
      height="12"
      viewBox="0 0 6 12"
      width="6"
    />
    <svg
      fill="#C5C5D6"
      height="12"
      viewBox="6 0 6 12"
      width="6"
    />
    <svg
      fill="#C5C5D6"
      height="12"
      viewBox="0 0 12 12"
      width="12"
    />
    <svg
      fill="#C5C5D6"
      height="12"
      viewBox="0 0 12 12"
      width="12"
    />
    <svg
      fill="#C5C5D6"
      height="12"
      viewBox="0 0 12 12"
      width="12"
    />
    <svg
      fill="#C5C5D6"
      height="12"
      viewBox="0 0 12 12"
      width="12"
    />
  </div>
  <div
    class="c0"
    height="12"
    readonly=""
    width="76"
  >
    <svg
      fill="#FFC24C"
      height="12"
      viewBox="0 0 12 12"
      width="12"
    />
    <svg
      fill="#C5C5D6"
      height="12"
      viewBox="0 0 12 12"
      width="12"
    />
    <svg
      fill="#C5C5D6"
      height="12"
      viewBox="0 0 12 12"
      width="12"
    />
    <svg
      fill="#C5C5D6"
      height="12"
      viewBox="0 0 12 12"
      width="12"
    />
    <svg
      fill="#C5C5D6"
      height="12"
      viewBox="0 0 12 12"
      width="12"
    />
  </div>
  <div
    class="c0"
    height="12"
    readonly=""
    width="76"
  >
    <svg
      fill="#FFC24C"
      height="12"
      viewBox="0 0 12 12"
      width="12"
    />
    <svg
      fill="#FFC24C"
      height="12"
      viewBox="0 0 1.200000000000001 12"
      width="1.200000000000001"
    />
    <svg
      fill="#C5C5D6"
      height="12"
      viewBox="1.200000000000001 0 10.799999999999999 12"
      width="10.799999999999999"
    />
    <svg
      fill="#C5C5D6"
      height="12"
      viewBox="0 0 12 12"
      width="12"
    />
    <svg
      fill="#C5C5D6"
      height="12"
      viewBox="0 0 12 12"
      width="12"
    />
    <svg
      fill="#C5C5D6"
      height="12"
      viewBox="0 0 12 12"
      width="12"
    />
  </div>
  <div
    class="c0"
    height="12"
    readonly=""
    width="76"
  >
    <svg
      fill="#FFC24C"
      height="12"
      viewBox="0 0 12 12"
      width="12"
    />
    <svg
      fill="#FFC24C"
      height="12"
      viewBox="0 0 2.3999999999999995 12"
      width="2.3999999999999995"
    />
    <svg
      fill="#C5C5D6"
      height="12"
      viewBox="2.3999999999999995 0 9.600000000000001 12"
      width="9.600000000000001"
    />
    <svg
      fill="#C5C5D6"
      height="12"
      viewBox="0 0 12 12"
      width="12"
    />
    <svg
      fill="#C5C5D6"
      height="12"
      viewBox="0 0 12 12"
      width="12"
    />
    <svg
      fill="#C5C5D6"
      height="12"
      viewBox="0 0 12 12"
      width="12"
    />
  </div>
  <div
    class="c0"
    height="12"
    readonly=""
    width="76"
  >
    <svg
      fill="#FFC24C"
      height="12"
      viewBox="0 0 12 12"
      width="12"
    />
    <svg
      fill="#FFC24C"
      height="12"
      viewBox="0 0 3.6000000000000005 12"
      width="3.6000000000000005"
    />
    <svg
      fill="#C5C5D6"
      height="12"
      viewBox="3.6000000000000005 0 8.399999999999999 12"
      width="8.399999999999999"
    />
    <svg
      fill="#C5C5D6"
      height="12"
      viewBox="0 0 12 12"
      width="12"
    />
    <svg
      fill="#C5C5D6"
      height="12"
      viewBox="0 0 12 12"
      width="12"
    />
    <svg
      fill="#C5C5D6"
      height="12"
      viewBox="0 0 12 12"
      width="12"
    />
  </div>
  <div
    class="c0"
    height="12"
    readonly=""
    width="76"
  >
    <svg
      fill="#FFC24C"
      height="12"
      viewBox="0 0 12 12"
      width="12"
    />
    <svg
      fill="#FFC24C"
      height="12"
      viewBox="0 0 4.799999999999999 12"
      width="4.799999999999999"
    />
    <svg
      fill="#C5C5D6"
      height="12"
      viewBox="4.799999999999999 0 7.200000000000001 12"
      width="7.200000000000001"
    />
    <svg
      fill="#C5C5D6"
      height="12"
      viewBox="0 0 12 12"
      width="12"
    />
    <svg
      fill="#C5C5D6"
      height="12"
      viewBox="0 0 12 12"
      width="12"
    />
    <svg
      fill="#C5C5D6"
      height="12"
      viewBox="0 0 12 12"
      width="12"
    />
  </div>
  <div
    class="c0"
    height="12"
    readonly=""
    width="76"
  >
    <svg
      fill="#FFC24C"
      height="12"
      viewBox="0 0 12 12"
      width="12"
    />
    <svg
      fill="#FFC24C"
      height="12"
      viewBox="0 0 6 12"
      width="6"
    />
    <svg
      fill="#C5C5D6"
      height="12"
      viewBox="6 0 6 12"
      width="6"
    />
    <svg
      fill="#C5C5D6"
      height="12"
      viewBox="0 0 12 12"
      width="12"
    />
    <svg
      fill="#C5C5D6"
      height="12"
      viewBox="0 0 12 12"
      width="12"
    />
    <svg
      fill="#C5C5D6"
      height="12"
      viewBox="0 0 12 12"
      width="12"
    />
  </div>
  <div
    class="c0"
    height="12"
    readonly=""
    width="76"
  >
    <svg
      fill="#FFC24C"
      height="12"
      viewBox="0 0 12 12"
      width="12"
    />
    <svg
      fill="#FFC24C"
      height="12"
      viewBox="0 0 7.200000000000001 12"
      width="7.200000000000001"
    />
    <svg
      fill="#C5C5D6"
      height="12"
      viewBox="7.200000000000001 0 4.799999999999999 12"
      width="4.799999999999999"
    />
    <svg
      fill="#C5C5D6"
      height="12"
      viewBox="0 0 12 12"
      width="12"
    />
    <svg
      fill="#C5C5D6"
      height="12"
      viewBox="0 0 12 12"
      width="12"
    />
    <svg
      fill="#C5C5D6"
      height="12"
      viewBox="0 0 12 12"
      width="12"
    />
  </div>
  <div
    class="c0"
    height="12"
    readonly=""
    width="76"
  >
    <svg
      fill="#FFC24C"
      height="12"
      viewBox="0 0 12 12"
      width="12"
    />
    <svg
      fill="#FFC24C"
      height="12"
      viewBox="0 0 8.399999999999999 12"
      width="8.399999999999999"
    />
    <svg
      fill="#C5C5D6"
      height="12"
      viewBox="8.399999999999999 0 3.6000000000000005 12"
      width="3.6000000000000005"
    />
    <svg
      fill="#C5C5D6"
      height="12"
      viewBox="0 0 12 12"
      width="12"
    />
    <svg
      fill="#C5C5D6"
      height="12"
      viewBox="0 0 12 12"
      width="12"
    />
    <svg
      fill="#C5C5D6"
      height="12"
      viewBox="0 0 12 12"
      width="12"
    />
  </div>
  <div
    class="c0"
    height="12"
    readonly=""
    width="76"
  >
    <svg
      fill="#FFC24C"
      height="12"
      viewBox="0 0 12 12"
      width="12"
    />
    <svg
      fill="#FFC24C"
      height="12"
      viewBox="0 0 9.600000000000001 12"
      width="9.600000000000001"
    />
    <svg
      fill="#C5C5D6"
      height="12"
      viewBox="9.600000000000001 0 2.3999999999999995 12"
      width="2.3999999999999995"
    />
    <svg
      fill="#C5C5D6"
      height="12"
      viewBox="0 0 12 12"
      width="12"
    />
    <svg
      fill="#C5C5D6"
      height="12"
      viewBox="0 0 12 12"
      width="12"
    />
    <svg
      fill="#C5C5D6"
      height="12"
      viewBox="0 0 12 12"
      width="12"
    />
  </div>
  <div
    class="c0"
    height="12"
    readonly=""
    width="76"
  >
    <svg
      fill="#FFC24C"
      height="12"
      viewBox="0 0 12 12"
      width="12"
    />
    <svg
      fill="#FFC24C"
      height="12"
      viewBox="0 0 10.799999999999999 12"
      width="10.799999999999999"
    />
    <svg
      fill="#C5C5D6"
      height="12"
      viewBox="10.799999999999999 0 1.200000000000001 12"
      width="1.200000000000001"
    />
    <svg
      fill="#C5C5D6"
      height="12"
      viewBox="0 0 12 12"
      width="12"
    />
    <svg
      fill="#C5C5D6"
      height="12"
      viewBox="0 0 12 12"
      width="12"
    />
    <svg
      fill="#C5C5D6"
      height="12"
      viewBox="0 0 12 12"
      width="12"
    />
  </div>
  <div
    class="c0"
    height="12"
    readonly=""
    width="76"
  >
    <svg
      fill="#FFC24C"
      height="12"
      viewBox="0 0 12 12"
      width="12"
    />
    <svg
      fill="#FFC24C"
      height="12"
      viewBox="0 0 12 12"
      width="12"
    />
    <svg
      fill="#C5C5D6"
      height="12"
      viewBox="0 0 12 12"
      width="12"
    />
    <svg
      fill="#C5C5D6"
      height="12"
      viewBox="0 0 12 12"
      width="12"
    />
    <svg
      fill="#C5C5D6"
      height="12"
      viewBox="0 0 12 12"
      width="12"
    />
  </div>
  <div
    class="c0"
    height="12"
    readonly=""
    width="76"
  >
    <svg
      fill="#FFC24C"
      height="12"
      viewBox="0 0 12 12"
      width="12"
    />
    <svg
      fill="#FFC24C"
      height="12"
      viewBox="0 0 12 12"
      width="12"
    />
    <svg
      fill="#FFC24C"
      height="12"
      viewBox="0 0 6 12"
      width="6"
    />
    <svg
      fill="#C5C5D6"
      height="12"
      viewBox="6 0 6 12"
      width="6"
    />
    <svg
      fill="#C5C5D6"
      height="12"
      viewBox="0 0 12 12"
      width="12"
    />
    <svg
      fill="#C5C5D6"
      height="12"
      viewBox="0 0 12 12"
      width="12"
    />
  </div>
  <div
    class="c0"
    height="12"
    readonly=""
    width="76"
  >
    <svg
      fill="#FFC24C"
      height="12"
      viewBox="0 0 12 12"
      width="12"
    />
    <svg
      fill="#FFC24C"
      height="12"
      viewBox="0 0 12 12"
      width="12"
    />
    <svg
      fill="#FFC24C"
      height="12"
      viewBox="0 0 12 12"
      width="12"
    />
    <svg
      fill="#C5C5D6"
      height="12"
      viewBox="0 0 12 12"
      width="12"
    />
    <svg
      fill="#C5C5D6"
      height="12"
      viewBox="0 0 12 12"
      width="12"
    />
  </div>
  <div
    class="c0"
    height="12"
    readonly=""
    width="76"
  >
    <svg
      fill="#FFC24C"
      height="12"
      viewBox="0 0 12 12"
      width="12"
    />
    <svg
      fill="#FFC24C"
      height="12"
      viewBox="0 0 12 12"
      width="12"
    />
    <svg
      fill="#FFC24C"
      height="12"
      viewBox="0 0 12 12"
      width="12"
    />
    <svg
      fill="#FFC24C"
      height="12"
      viewBox="0 0 6 12"
      width="6"
    />
    <svg
      fill="#C5C5D6"
      height="12"
      viewBox="6 0 6 12"
      width="6"
    />
    <svg
      fill="#C5C5D6"
      height="12"
      viewBox="0 0 12 12"
      width="12"
    />
  </div>
  <div
    class="c0"
    height="12"
    readonly=""
    width="76"
  >
    <svg
      fill="#FFC24C"
      height="12"
      viewBox="0 0 12 12"
      width="12"
    />
    <svg
      fill="#FFC24C"
      height="12"
      viewBox="0 0 12 12"
      width="12"
    />
    <svg
      fill="#FFC24C"
      height="12"
      viewBox="0 0 12 12"
      width="12"
    />
    <svg
      fill="#FFC24C"
      height="12"
      viewBox="0 0 12 12"
      width="12"
    />
    <svg
      fill="#C5C5D6"
      height="12"
      viewBox="0 0 12 12"
      width="12"
    />
  </div>
  <div
    class="c0"
    height="12"
    readonly=""
    width="76"
  >
    <svg
      fill="#FFC24C"
      height="12"
      viewBox="0 0 12 12"
      width="12"
    />
    <svg
      fill="#FFC24C"
      height="12"
      viewBox="0 0 12 12"
      width="12"
    />
    <svg
      fill="#FFC24C"
      height="12"
      viewBox="0 0 12 12"
      width="12"
    />
    <svg
      fill="#FFC24C"
      height="12"
      viewBox="0 0 12 12"
      width="12"
    />
    <svg
      fill="#FFC24C"
      height="12"
      viewBox="0 0 6 12"
      width="6"
    />
    <svg
      fill="#C5C5D6"
      height="12"
      viewBox="6 0 6 12"
      width="6"
    />
  </div>
</div>
`;
